newIDE/electron-app/app/main.js

```diff
@@ -129,6 +133,13 @@ app.on('ready', function() {
     }
   });
 
+  mainWindow.webContents.on('devtools-reload-page', () => {
```

Comment:
I would rather avoid relying on devtools events, and also this is not using [`LocalGDJSDevelopmentWatcher` ](https://github.com/4ian/GDevelop/blob/master/newIDE/app/src/GameEngineFinder/LocalGDJSDevelopmentWatcher.js) which is a bit sad because there is now two places where the editor is magically launch a npm script (there and in LocalGDJSDevelopmentWatcher - one using npm, the other using node).

Finally, I prefer to keep the sources of the electron-app as lean as possible. 

So I think we could instead of this addition modify `LocalGDJSDevelopmentWatcher` so that it's re-importing GDJS (and extensions) anytime it's being set up (i.e: when the app is launched in development mode).

For this, make a call to `importGDJSRuntime` (`importGDJSRuntime().catch(() => {});`) in:

https://github.com/4ian/GDevelop/blob/d0247c7df33e5a73049be738831ca6bfbe289a5d/newIDE/app/src/GameEngineFinder/LocalGDJSDevelopmentWatcher.js#L177-L184

Probably just after the "first return". React will run the code in "useEffect" when the array passed to useEffect (`[shouldWatch]`) is changed, i.e: when the preference is changed in the preferences dialog, or when the component is mounted (i.e: at the start of the app).

Reply:
This will actually make the watcher more "robust", as this will guarantee that if you deactivated the watcher for a while, the sources are re-imported as soon as your re-activate it :)

Reply:
thank you, works like a charm :)